src/server/location/location.service.ts

```diff
@@ -280,4 +281,41 @@ export class LocationService {
     };
     return result;
   }
+
+  /**
+   * Updates an existing room's name and capacity information.
+   * The campus and building cannot be updated.
+   */
+  public async updateRoom(roomId: string, roomInfo: UpdateRoom)
+    : Promise<RoomAdminResponse> {
+    let existingRoom: Room;
+    try {
+      existingRoom = await this.roomRepository.findOneOrFail(roomId, {
+        relations: ['building'],
+      });
+    } catch (e) {
+      if (e instanceof EntityNotFoundError) {
+        throw new NotFoundException(`Unable to find room ${roomId}`);
```

Comment:
The controller method is expecting an `EntityNotFoundError` and will convert that into a `NotFoundException`, so I think we're actually fine to let the error bubble up to there instead of catching and rethrowing it.

Reply:
Ah good point! This was changed in commit [09698b6](https://github.com/seas-computing/course-planner/pull/581/commits/09698b6203bfed2e45c4e9c6e17a31947bfb365d) so that the service is just letting the error bubble up to the controller instead.